### DISCUSSION TITLE
How do I get the type for Next/Image?

### DISCUSSION THREAD

**Discussion by U-4-E-A** [vercel/next.js#43738] [Help] [answered]
```
import React, { ReactElement } from 'react'
import Image from 'next/image'
const NextResponsiveImage = (props: typeof Image): ReactElement => {
const { className, ...rest } = props
return (
    [...]
  )
}
export default NextResponsiveImage
```
With the code above, I get the error: -

`Property 'className' does not exist on type 'ForwardRefExoticComponent<Omit<DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>, "height" | ... 5 more ... | "srcSet"> & { ...; } & RefAttributes<...>>'.ts(2339)`

What am I missing here? Thanks.

**Comment by icyJoseph** [comment, marked as answer]
You wanna do something like this:

```tsx
import Image, { type ImageProps } from "next/image";

// I'd recommend skipping typing
// the return explicitly, for certain cases
// it is best to let type inference do it's job
const NextResponsiveImage = (props: ImageProps) => {
  const { className, ...rest } = props;
  return <Image {...rest} />;
};
```

Get the props from `next/image` directly.

## Alternative

Sometimes a framework might not expose types, then you could:

```tsx
import Image from "next/image";

const NextResponsiveImage = (props: Parameters<typeof Image>[0]) => {
  const { className, ...rest } = props;
  return <Image {...rest} />;
};
```

- https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype

**Reply by U-4-E-A** [reply]
Thank you!
